src/NATS.Client.JetStream/Models/StreamCreateRequest.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Text.Json.Serialization;
+
+namespace NATS.Client.JetStream.Models;
+
+public record StreamCreateRequest
+{
+    [Required]
+    [JsonPropertyName("name")]
+    public string Name { get; set; }
+
+    [Required]
+    [JsonPropertyName("subjects")]
+    public string[] Subjects { get; set; }
+
+    [JsonPropertyName("retention")]
+    public string Retention { get; set; } = "limits";
+
+    [JsonPropertyName("max_consumers")]
+    public long MaxConsumers { get; set; } = -1;
+
+    [JsonPropertyName("max_msgs_per_subject")]
+    public long MaxMsgsPerSubject { get; set; } = -1;
+
+    [JsonPropertyName("max_msgs")]
+    public long MaxMsgs { get; set; } = -1;
+
+    [JsonPropertyName("max_bytes")]
+    public long MaxBytes { get; set; } = -1;
+
+    [JsonPropertyName("max_age")]
+    public long MaxAge { get; set; } = 0;
+
+    [JsonPropertyName("max_msg_size")]
+    public long MaxMsgSize { get; set; } = -1;
+
+    [JsonPropertyName("storage")]
+    public string Storage { get; set; } = "file";
+
+    [JsonPropertyName("discard")]
+    public string Discard { get; set; } = "old";
+
+    [JsonPropertyName("num_replicas")]
+    public long NumReplicas { get; set; } = 1;
+
+    [JsonPropertyName("duplicate_window")]
+    public long DuplicateWindow { get; set; } = 120000000000;
```

Comment:
When sending to server, do not set defaults except for very few things. This allows the server to control what defaults are. This includes booleans. Values should not be sent to the server if they are not explicitly set by the user or are specific defaults.

Reply:
defaults:
RetentionPolicy = limits
StorageType = file
DiscardPolicy = old